terraform/production/datamart_imported_logs.tf

```diff
@@ -0,0 +1,52 @@
+# Logs imported from Datamart for retention
+
+# At time of writing our policy is to retain production logs for two years,
+# so this should be kept until March 2025, after which it can be deleted
+module "datamart_imported_logs_bucket" {
+  source = "../modules/s3_bucket"
+
+  bucket_name                   = "dluhc-datamart-imported-logs-production"
+  access_log_bucket_name        = "dluhc-datamart-imported-logs-production-access-logs"
+  access_s3_log_expiration_days = local.s3_log_expiration_days
+  policy                        = data.aws_iam_policy_document.allow_access_from_datamart.json
+  restrict_public_buckets       = false
+}
+
+locals {
+  datamart_account_id = "090682378586"
+}
+
+data "aws_iam_policy_document" "allow_access_from_datamart" {
+  statement {
+    principals {
+      type        = "AWS"
+      identifiers = [local.datamart_account_id]
+    }
+
+    actions = [
+      "s3:GetBucketLocation", "s3:GetEncryptionConfiguration", "s3:ListBucket", "s3:GetObject", "s3:DeleteObject"
+    ]
+
+    resources = [
+      module.datamart_imported_logs_bucket.bucket_arn,
+      "${module.datamart_imported_logs_bucket.bucket_arn}/*",
+    ]
+  }
+
+  statement {
+    principals {
+      type        = "AWS"
+      identifiers = [local.datamart_account_id]
+    }
+
+    actions = ["s3:PutObject"]
+
+    resources = ["${module.datamart_imported_logs_bucket.bucket_arn}/*"]
+
+    condition {
+      test     = "StringEquals"
+      values   = ["GLACIER"] # "Glacier Flexible Retrieval"
```

Comment:
The ticket says "Glacier Deep Archive", do we want that instead? If there transferring loads of data then probably yes, it's just inconvenient if we ever want to use it.

Reply:
Would be `"DEEP_ARCHIVE"`, likely worth it if we're getting >>10TB of logs

Reply:
I really don't expect we'll need the logs given that we migrated almost 6 months ago. We're just making sure we're meeting compliance requirements before we tell Datamart to finish killing everything Delta-related. 

But sure, if there's not much to be saved for <10TB, we can use "GLACIER". They definitely have >10TB of logs, but I think mostly logs that we're not keeping.